draft-ietf-quic-invariants.md

```diff
@@ -150,10 +150,25 @@ version-specific semantics are marked with an X.
 {: #fig-long title="QUIC Long Header"}
 
 A QUIC packet with a long header has the high bit of the first octet set to 1.
+All other bits in that octet are version specific.
+
+The next octet contains the length in octets of the two Connection IDs (see
+{{connection-id}}) that follow.  Each length is encoded as a 4 bit unsigned
+integer.  The length of the Destination Connection ID occupies the high bits of
+the octet and the length of the Source Connection ID occupying the low bits of
+the octet.  An encoded length of 0 indicates that the connection ID is also 0
```

Comment:
I think there are valid use cases for shorter connection IDs(ie: 4 bytes in particular), so I'd be biased towards using 5 bits for each length.  I'm not sure if I'd want to overlap this with the first byte to reduce the number of bits we'd want to grease or just deal with randomizing 3 bits in each byte.

Reply:
+1 to Ian here.

Reply:
I actually think that shorter connection IDs is harmful.  If we are to encourage the use of NEW_CONNECTION_ID, the space needs to be large enough to make the use of a new identifier meaningful.